src/main/objects/RulesChecker.js

```diff
@@ -136,9 +140,9 @@ function checkConnection(prevChord, currentChord){
             }
             if (dominantVoiceWith3 > -1 &&
                 !prevChord.notes[dominantVoiceWith3].equalPitches(currentChord.notes[dominantVoiceWith3]) &&
-                !Utils.containsChordComponent(currentChord.harmonicFunction.omit, "1") &&
-                !currentChord.notes[dominantVoiceWith3].chordComponentEquals("1") &&
-                !currentChord.notes[dominantVoiceWith3].chordComponentEquals("7")) return -1;
+                !Utils.containsBaseChordComponent(currentChord.harmonicFunction.omit, "1") &&
+                !currentChord.notes[dominantVoiceWith3].baseChordComponentEquals("1") &&
+                !currentChord.notes[dominantVoiceWith3].baseChordComponentEquals("7")) return -1;
```

Comment:
Myślę, że nie powinniśmy mnożyć funckji typu equals. Lepiej zawołać odpowiednią funckję z ChordComponent. Nie wykluczam że coś jest nie tak z equals na chordComponent. Daj znać jeśli tak

Reply:
To są funkcje powstałe z refactoru rule checkera

Reply:
Ze względu na akordy obniżone należy porównać tu baseChordComponenty (doszło 1> np)